mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
```

Comment:
The function parameter `id` is not validated. It should be checked if it's a valid ObjectId before calling `Product.findById(id)`.